### ISSUE TITLE
Passing invalid URL to Picasso

### ISSUE THREAD

**Issue by harkin** [intercom/intercom-android#172]
taken from https://github.com/intercom/intercom-android/issues/85

> We are running on build 1.1.9 and are seeing this crash happen still. I just updated the library to 1.1.12 and have our app ready for beta on Monday, but I see no fixes for this between 1.1.9-1.1.12. I think this issue still exists, or something new was introduced.
> report.txt
```
Stack Trace
_________________________________
0	java.lang.RuntimeException: Could not dispatch event: class io.intercom.android.sdk.models.Events.ConversationEvent to handler [EventHandler public void io.intercom.android.sdk.conversation.ConversationFragment.conversationSuccess(io.intercom.android.sdk.models.Events.ConversationEvent)]: Path must not be empty.
1		at io.intercom.com.squareup.otto.Bus.throwRuntimeException(Bus.java:460)
2		at io.intercom.com.squareup.otto.Bus.dispatch(Bus.java:387)
3		at io.intercom.com.squareup.otto.Bus.dispatchQueuedEvents(Bus.java:368)
4		at io.intercom.com.squareup.otto.Bus.post(Bus.java:337)
5		at io.intercom.android.sdk.api.MainThreadBus.post(MainThreadBus.java:23)
6		at io.intercom.android.sdk.api.CallbackHolder$5.onSuccess(CallbackHolder.java:80)
7		at io.intercom.android.sdk.api.CallbackHolder$5.onSuccess(CallbackHolder.java:78)
8		at io.intercom.android.sdk.api.BaseCallback.success(BaseCallback.java:21)
9		at io.intercom.retrofit.CallbackRunnable$1.run(CallbackRunnable.java:45)
10		at android.os.Handler.handleCallback(Handler.java:739)
11		at android.os.Handler.dispatchMessage(Handler.java:95)
12		at android.os.Looper.loop(Looper.java:145)
13		at android.app.ActivityThread.main(ActivityThread.java:6897)
14		at java.lang.reflect.Method.invoke(Native Method)
15		at java.lang.reflect.Method.invoke(Method.java:372)
16		at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
17		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
18	Caused by: java.lang.IllegalArgumentException: Path must not be empty.
19		at io.intercom.com.squareup.picasso.Picasso.load(Picasso.java:297)
20		at io.intercom.android.sdk.blocks.NetworkImage.addImage(NetworkImage.java:45)
21		at io.intercom.android.sdk.blocks.BlockType$7.generateView(BlockType.java:45)
22		at io.intercom.android.sdk.blocks.Blocks.createBlocks(Blocks.java:32)
23		at io.intercom.android.sdk.conversation.BaseConversationFragment.createMessageUI(BaseConversationFragment.java:120)
24		at io.intercom.android.sdk.conversation.ConversationFragment.createConversationUI(ConversationFragment.java:688)
25		at io.intercom.android.sdk.conversation.ConversationFragment.setupWithConversation(ConversationFragment.java:640)
26		at io.intercom.android.sdk.conversation.ConversationFragment.conversationSuccess(ConversationFragment.java:478)
27		at java.lang.reflect.Method.invoke(Native Method)
28		at java.lang.reflect.Method.invoke(Method.java:372)
29		at io.intercom.com.squareup.otto.EventHandler.handleEvent(EventHandler.java:89)
30		at io.intercom.com.squareup.otto.Bus.dispatch(Bus.java:385)
31		... 15 more
32	java.lang.IllegalArgumentException: Path must not be empty.
33		at io.intercom.com.squareup.picasso.Picasso.load(Picasso.java:297)
34		at io.intercom.android.sdk.blocks.NetworkImage.addImage(NetworkImage.java:45)
35		at io.intercom.android.sdk.blocks.BlockType$7.generateView(BlockType.java:45)
36		at io.intercom.android.sdk.blocks.Blocks.createBlocks(Blocks.java:32)
37		at io.intercom.android.sdk.conversation.BaseConversationFragment.createMessageUI(BaseConversationFragment.java:120)
38		at io.intercom.android.sdk.conversation.ConversationFragment.createConversationUI(ConversationFragment.java:688)
39		at io.intercom.android.sdk.conversation.ConversationFragment.setupWithConversation(ConversationFragment.java:640)
40		at io.intercom.android.sdk.conversation.ConversationFragment.conversationSuccess(ConversationFragment.java:478)
41		at java.lang.reflect.Method.invoke(Native Method)
42		at java.lang.reflect.Method.invoke(Method.java:372)
43		at io.intercom.com.squareup.otto.EventHandler.handleEvent(EventHandler.java:89)
44		at io.intercom.com.squareup.otto.Bus.dispatch(Bus.java:385)
45		at io.intercom.com.squareup.otto.Bus.dispatchQueuedEvents(Bus.java:368)
46		at io.intercom.com.squareup.otto.Bus.post(Bus.java:337)
47		at io.intercom.android.sdk.api.MainThreadBus.post(MainThreadBus.java:23)
48		at io.intercom.android.sdk.api.CallbackHolder$5.onSuccess(CallbackHolder.java:80)
49		at io.intercom.android.sdk.api.CallbackHolder$5.onSuccess(CallbackHolder.java:78)
50		at io.intercom.android.sdk.api.BaseCallback.success(BaseCallback.java:21)
51		at io.intercom.retrofit.CallbackRunnable$1.run(CallbackRunnable.java:45)
52		at android.os.Handler.handleCallback(Handler.java:739)
53		at android.os.Handler.dispatchMessage(Handler.java:95)
54		at android.os.Looper.loop(Looper.java:145)
55		at android.app.ActivityThread.main(ActivityThread.java:6897)
56		at java.lang.reflect.Method.invoke(Native Method)
57		at java.lang.reflect.Method.invoke(Method.java:372)
58		at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
59		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
```

**Comment by harkin** [comment]
@lithid I've found the issue in the library and it'll be patched in our next release. We weren't handling lack of data well. I'm a little concerned that we had this lack of data though, do you happen to have an example of a conversation that was crashing so that I could track what's going on through our backend systems?

**Comment by bsiegel** [comment]
@harkin Jumping in for @lithid, I can send you a link to the user who experienced the crashes. A total of three different support sessions were created since each time the crash occurred and the user re-engaged us, it started a new support session. Do you have an e-mail address I can send this info to?

**Comment by harkin** [comment]
sure if you could send it on to henry.larkin(at)intercom.mail.intercom.io that'd be perfect!

**Comment by harkin** [comment]
This one is fixed in 1.1.13 now
